frontend/public/components/modals/configure-cluster-upstream-modal.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+
+import { ClusterVersionModel } from '../../models';
+import { ClusterVersionKind, k8sPatch } from '../../module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '../factory/modal';
+import { ExternalLink, HandlePromiseProps, withHandlePromise } from '../utils';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { RadioInput } from '../radio';
+import {
+  CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL,
+  CLUSTER_VERSION_CUSTOM_UPSTREAM_SERVER_URL_PLACEHOLDER,
+} from '@console/shared/src/constants';
+import { TextInput } from '@patternfly/react-core';
+
+export const ConfigureClusterUpstreamModal = withHandlePromise(
+  (props: ConfigureClusterUpstreamModalProps) => {
+    const { cv, handlePromise, close } = props;
+    const currentUpstream = cv?.spec?.upstream;
+
+    const [selectedRadio, setSelectedRadio] = React.useState(
+      currentUpstream ? 'custom' : 'default',
+    );
+    const [customUrl, setCustomUrl] = React.useState(currentUpstream);
```

Comment:
nit: Current naming conventions would make this `customURL`

```suggestion
    const [customURL, setCustomURL] = React.useState(currentUpstream);
```